src/lightning/data/streaming/writer.py

```diff
@@ -29,11 +29,35 @@
     from torch.utils._pytree import PyTree, tree_flatten, treespec_dumps
 
 
-def _get_data_optimizer_node_rank() -> Optional[int]:
-    node_rank = os.getenv("DATA_OPTIMIZER_NODE_RANK", None)
-    if node_rank is not None:
-        return int(node_rank)
-    return node_rank
+_FORMAT_TO_RATIO = {
+    "kb": 1024,
+    "mb": 1024**2,
+    "gb": 1024**3,
+    "tb": 1024**4,
```

Comment:
It's probably unexpected to have the unit not match the number of bytes here.
1 Kilobyte would be 1000 bytes, not 1024.